lib/Runtime/PlatformAgnostic/Platform/Common/Intl.cpp

```diff
@@ -121,6 +124,23 @@ class Output
         }                                                                     \
     } while (false)
 
+#define ICU_ASSERT_GOTO(e, expr, l)                                           \
+    do                                                                        \
+    {                                                                         \
+        if (U_FAILURE(e))                                                     \
+        {                                                                     \
+            ICU_DEBUG_PRINT(ICU_ERROR_FMT, u_errorName(e));                   \
+            AssertMsg(false, u_errorName(e));                                 \
+            goto l;                                                           \
```

Comment:
Here *probably* doesn't need a failfast, since the intent is to recover by going to the label specified if the Assert didn't stop execution (debug builds).